### ISSUE TITLE
References are not resolved from transitive dependencies (of more than one level)

### ISSUE THREAD

**Issue by SimonCockx** [xmlet/XsdParser#58]
**Describe the bug**
Suppose I have file `A.xsd` which includes `B.xsd` which includes `C.xsd` which includes `D.xsd`. If I have a reference to `D.xsd` in `A.xsd`, it will not be resolved by the parser.

This is because transitive dependencies are implemented incorrectly: they are not implemented in a recursive way and will only include "one level" of transitivity, e.g., `C.xsd` in my example will be included, but `D.xsd` will not.

See the [`XsdParserCore`](https://github.com/xmlet/XsdParser/blob/master/src/main/java/org/xmlet/xsdparser/core/XsdParserCore.java#L275) class:
```
Set<String> transitiveIncludes = new HashSet<>();

for(String includedFile : includedFiles){
    parseElements.keySet()
             .stream()
             .filter(fileNameAux -> fileNameAux.endsWith(includedFile))
             .findFirst()
             .ifPresent(fullIncludedFileName -> transitiveIncludes.addAll(parseElements.get(fullIncludedFileName)
                  .stream()
                  .filter(referenceBase -> referenceBase instanceof ConcreteElement && referenceBase.getElement() instanceof XsdInclude)
                  .map(referenceBase -> (((XsdInclude) referenceBase.getElement()).getSchemaLocation()))
                  .collect(Collectors.toList())));
}
```

**Expected behavior**
I expect transitive dependencies to work correctly, i.e., transitively.

**Library Version**
1.2.6

**Comment by darthweiter** [comment]
This is a nice solution.
I found an other bug, the unsolved references are not solved correctly.
I used your solution for the transitve includes there and it works -> https://github.com/xmlet/XsdParser/pull/61

**Comment by lcduarte** [comment]
Thanks for the issue and the PR, it looks good! I've approved it, I'll review the one opened by @darthweiter since it appear to be related. I may deploy a new version with the suggested solutions today if everything works.

**Comment by lcduarte** [comment]
I've released a new version, [1.2.8](https://github.com/xmlet/XsdParser/tree/xsdParser-1.2.8), with the code for this issue.
